src/fa/game_session.py

```diff
@@ -94,6 +94,16 @@ def listen(self):
         else:
             self.connectivity.prepare()
 
+    def _needs_game_connection(fn):
+        def wrap(self, *args, **kwargs):
+            if self._game_connection is None:
+                self._logger.warn("{}.{}: tried to run without a game connection".format(
+                    self.__class__.__name__, fn.__name__))
+            else:
+                return fn(self, *args, **kwargs)
+        return wrap
+
+    @_needs_game_connection
```

Comment:
I don't know what that means - maybe pycharm doesn't like that this decorator can be treated as a method.